src/java.base/share/classes/sun/launcher/LauncherHelper.java

```diff
@@ -920,58 +910,65 @@ private static boolean doesExtendFXApplication(Class<?> mainClass) {
         return false;
     }
 
-    private static boolean isStaticMain = false;
-    private static boolean noArgMain = false;
-
-    // Check the existence and signature of main and abort if incorrect.
-    private static void validateMainMethod(Class<?> mainClass) {
+    // Determine the presence of a valid main method on the MainEntry.Builder.mainClass.
+    // abort() if validation fails.
+    private static MainEntry requireValidMainMethod(MainEntry.Builder builder) {
```

Comment:
I think the `MainEntry.Builder` class is redundant - we should just pass in the local variables here.